src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -587,7 +587,159 @@ public static void Main()
 
             var program = (NamedTypeSymbol)comp.GetMember("Program");
             var attributeData = (SourceAttributeData)program.GetAttributes()[0];
-            Assert.Equal(new[] { 0, -1 }, attributeData.ConstructorArgumentsSourceIndices);
+            Assert.True(attributeData.ConstructorArgumentsSourceIndices.IsDefault);
+
+            var attributeSyntax = (AttributeSyntax)attributeData.ApplicationSyntaxReference.GetSyntax();
+            Assert.Equal("a: true", attributeData.GetAttributeArgumentSyntax(parameterIndex: 0, attributeSyntax).ToString());
+            Assert.Equal(@"b: new object[] { ""Hello"", ""World"" }", attributeData.GetAttributeArgumentSyntax(parameterIndex: 1, attributeSyntax).ToString());
+        }
+
+        [Fact]
+        public void TestCallWithOptionalParametersInsideAttribute()
+        {
+            var source = @"
+using System;
+
+class Attr : Attribute { public Attr(int x) { } }
+
+class C
+{
+    [Attr(M())]
+    void M0() { }
+
+    public static int M(int x = 0) => x;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,11): error CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type
+                //     [Attr(M())]
+                Diagnostic(ErrorCode.ERR_BadAttributeArgument, "M()").WithLocation(8, 11));
+
+            var m0 = comp.GetMember<MethodSymbol>("C.M0");
+            var attrs = m0.GetAttributes();
+            Assert.Equal(1, attrs.Length);
+            Assert.Equal(TypedConstantKind.Error, attrs[0].ConstructorArguments.Single().Kind);
+        }
+
+        [Fact]
+        public void TestAttributeCallerInfoSemanticModel()
+        {
+            var source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class Attr : Attribute { public Attr([CallerMemberName] string s = null) { } }
+
+class C
+{
+    [Attr()]
+    void M0() { }
+}
+";
```

Comment:
>" [](http://example.com/codeflow?start=0&length=1)

Consider verifying the member name is emitted as the attribute constructor argument.